src/stratigraphy/evaluation/groundwater_evaluator.py

```diff
@@ -0,0 +1,128 @@
+
+        Returns:
+            OverallGroundwaterMetrics: The overall groundwater metrics.
+        """
+        overall_groundwater_metrics = OverallGroundwaterMetrics()
+
+        for groundwater_in_doc in self.groundwater_entries:
+            filename = groundwater_in_doc.filename
+            ground_truth_data = self.groundwater_ground_truth.for_file(filename)
+            if ground_truth_data is None or ground_truth_data.get("groundwater") is None:
+                ground_truth = []  # If no ground truth is available, set it to an empty list
+            else:
+                ground_truth = ground_truth_data.get("groundwater")
+
+            ############################################################################################################
+            ### Compute the metadata correctness for the groundwater information.
+            ############################################################################################################
+            gt_groundwater = [
+                Groundwater.from_json_values(
+                    depth=json_gt_data.get("depth"),
+                    date=json_gt_data.get("date"),
+                    elevation=json_gt_data.get("elevation"),
+                )
+                for json_gt_data in ground_truth
+            ]
+
+            groundwater_metrics = count_against_ground_truth(
+                [
+                    (
+                        entry.feature.depth,
+                        entry.feature.format_date(),
+                        entry.feature.elevation,
+                    )
+                    for entry in groundwater_in_doc.groundwater
+                ],
+                [(entry.depth, entry.format_date(), entry.elevation) for entry in gt_groundwater],
+            )
+            groundwater_depth_metrics = count_against_ground_truth(
+                [entry.feature.depth for entry in groundwater_in_doc.groundwater],
+                [entry.depth for entry in gt_groundwater],
+            )
+            groundwater_elevation_metrics = count_against_ground_truth(
+                [entry.feature.elevation for entry in groundwater_in_doc.groundwater],
+                [entry.elevation for entry in gt_groundwater],
+            )
+            groundwater_date_metrics = count_against_ground_truth(
+                [entry.feature.format_date() for entry in groundwater_in_doc.groundwater],
+                [entry.format_date() for entry in gt_groundwater],
+            )
+
+            file_groundwater_metrics = GroundwaterMetrics(
+                groundwater_metrics=groundwater_metrics,
+                groundwater_depth_metrics=groundwater_depth_metrics,
+                groundwater_elevation_metrics=groundwater_elevation_metrics,
+                groundwater_date_metrics=groundwater_date_metrics,
+                filename=filename,
+            )  # TODO: This clashes with the OverallMetrics object
```

Comment:
how do you mean this exactly, what is the "clash" that you are seeing?

Reply:
I was not super happy with the fact that I could not directly re-use the `OverallMetrics` class and had to create this new class as we also wanted to unify the evaluation across the different objects. However, I think it could be worse. If you wish, we can open a ticket to look into this. I do believe this ticket is a bit out of scope.